File: src/org/fhcrc/matsen/startree/spark/StarTreeKryoRegistrator.java
package org.fhcrc.matsen.startree.spark;

import cern.colt.matrix.impl.DenseDoubleMatrix2D;
import com.esotericsoftware.kryo.Kryo;
import dr.evolution.alignment.SimpleAlignment;
import org.fhcrc.matsen.startree.TwoTaxonResult;

/**
 * Created by cmccoy on 12/9/13.
 */
public class StarTreeKryoRegistrator implements org.apache.spark.serializer.KryoRegistrator {
    @Override
    public void registerClasses(Kryo kryo) {
        kryo.register(DenseDoubleMatrix2D.class);
        kryo.register(cern.colt.matrix.impl.RCDoubleMatrix2D.class);
        kryo.register(SimpleAlignment.class);
        kryo.register(TwoTaxonResult.class);
    }
}

File: src/main/scala/org/fhcrc/matsen/startree/spark/StarTreeKryoRegistrator.scala
package org.fhcrc.matsen.startree.spark

import com.esotericsoftware.kryo.Kryo
import dr.evolution.alignment.SimpleAlignment
import org.fhcrc.matsen.startree.TwoTaxonResult
import org.apache.commons.math.linear.{ArrayRealVector,BlockRealMatrix}
import org.apache.spark.serializer.KryoRegistrator

/**
 * Created by cmccoy on 1/2/14.
 */
class StarTreeKryoRegistrator extends KryoRegistrator {
  // Classes to register with Kryo
  private[this] val classes = Seq(classOf[SimpleAlignment],
    classOf[ArrayRealVector],
    classOf[BlockRealMatrix],
    classOf[TwoTaxonResult])

  def registerClasses(kryo : Kryo) = {
    classes.foreach(kryo.register)
  }
}
